### ISSUE TITLE
Option to specifiy if query in Workspace Symbol request applies to name(default) or name and containerName

### ISSUE THREAD

**Issue by ramya-rao-a** [Microsoft/language-server-protocol#183]
See https://github.com/sourcegraph/go-langserver/issues/150#issuecomment-279758788

Currently, the Workspace Symbol request is expected to return symbol definitions in the workspace.
As mentioned in the comment linked above, it can be useful for the query to be applied to the "containerName" as well.

**Comment by felixfbecker** [comment]
Current behaviour in PHP for example is to just match the fully qualified symbol name, which contains `containerName`, `name` and more.

**Comment by chrmarti** [comment]
The UI (and user) might become overwhelmed by the number of results when including the container name. Also: The UI doesn't seem to have good way to present the tree structure of the results, which makes it harder to scan the results when including the container name.

**Comment by felixfbecker** [comment]
As a user when typing in `java.util.` I expect to have `ArrayList` among others as suggestion (where `containerName` is `java.util`). If the query only applied to name, I would get zero results. This example can be adapted for many languages

**Comment by chrmarti** [comment]
Good point, but would you expect the class members to be shown in the results too?

**Comment by felixfbecker** [comment]
Yes. If I type `MyClass::` in PHP I want to see all methods.

**Comment by chrmarti** [comment]
And if you have typed just `My`? There might be many classes with even more members with that prefix. Maybe only once you've chosen a particular class you also want to see the methods?

**Comment by dbaeumer** [comment]
I agree that we should not differentiate this in the API but rather let the server interpret the name since the server knows how containers are interpreted and how namespace are separated. So java.util.List* should show all List* classes in the java.util package.
